specreduce/core.py

```diff
@@ -2,3 +2,246 @@
+
+        # To ensure the ratio between the standard star flux and observed flux
+        # has proper flux units
+        ratio = np.abs(np.array(std_flux_ds, dtype="float") /
+                       np.array(obj_flux_ds, dtype="float"))
+
+        # Interpolate flux calibration (sensfunc) onto object's wavelength grid
+        # 3 modes of interpolation available: 'linear', 'cubic spline', and 'polynomial'
+
+        # To check if a mode of interpolation has been specified
+        if mode not in ("linear", "spline", "poly"):
+            mode = "spline"
+            print("Warning: No interpolation mode has been set"
+                  "Setting mode to the default 'spline'.")
+
+        # Fit the log of the above sensfunc ratio
+        # As IRAF uses 2.5*log(ratio), the quantities are in mag units
+        logsensfunc = np.log10(ratio)
+
+        # Interpolate the logsensfunc back onto the wavelength grid
+        if mode == "linear":
+            sensfunc = np.interp(obj_wave, obj_wave_ds, logsensfunc)
+        elif mode == "spline":
+            spline = UnivariateSpline(obj_wave_ds, logsensfunc, ext=0, k=2, s=0.0025)
+            sensfunc = spline(obj_wave)
+        elif mode == "poly":
+            fit = np.polyfit(obj_wave_ds, logsensfunc, polyorder)
+            sensfunc = np.polyval(fit, obj_wave)
+
+        if show is True:
+            plt.figure()
+            plt.plot(
+                std_wave, std_flux, "r", alpha=0.5, label="Standard flux"
+            )
+            plt.xlabel("Wavelength")
+            plt.ylabel("Standard flux")
+            plt.legend()
+            plt.show()
+
+            plt.figure()
+            plt.plot(
+                obj_wave, obj_flux, "k", label="Observed counts"
+            )
+            plt.plot(
+                obj_wave_ds, obj_flux_ds, "bo", label="Downsampling observed"
+            )
+            plt.xlabel("Wavelength")
+            plt.ylabel("Observed count rate")
+            plt.legend()
+            plt.show()
+
+            plt.figure()
+            plt.plot(
+                obj_wave_ds, logsensfunc, "ko", label="Sensfunc"
+            )
+            plt.plot(
+                obj_wave, sensfunc, label="Interpolated sensfunc"
+            )
+            plt.xlabel("Wavelength")
+            plt.ylabel("Log of sensfunc")
+            plt.legend()
+            plt.show()
+
+            plt.figure()
+            plt.plot(
+                obj_wave, obj_flux * (10 ** sensfunc), "k", label="Applied sensfunc"
+            )
+            plt.plot(
+                std_wave, std_flux, "ro", alpha=0.5, label="Standard flux"
+            )
+            plt.xlabel("Wavelength")
+            plt.ylabel("Standard flux")
+            plt.legend()
+            plt.show()
+
+        else:
+            sensfunc = np.zeros_like(obj_wave)
+            msg2 = "Warning: No valid standard star file has been entered"
+            print(msg2)
+
+            return 10**sensfunc
+
```

Comment:
It also might be worthwhile to discuss what might be the best object for this to return -- it might make sense to return something with error information so that can be propagated.

Reply:
Yeah, that would be better than returning some `10**np.zeros_like()` object which would give the false impression that flux calibration has indeed been successful.

Reply:
> this looks quite good, and i like the new class structure.
> 
> I find it very valuable (both for development and as a potential user) to have a demo notebook or file. I know there are astropy tutorials, but could/should we include one alongside this? i.e. I don't quite know how the API works in my modeled workflow yet

We could include examples in the `specreduce` docs. And moreover, I could start working on some tutorial notebooks on Learn Astropy regarding the `fluxcal` class as a start in parallel.